kymatio/scattering3d/backend/torch_backend.py

```diff
@@ -58,11 +58,129 @@ def compute_integrals(input_array, integral_powers):
     return integrals
 
 
+class Pad(object):
+    def __init__(self, pad_size, input_size, pre_pad=False):
+        """Padding which allows to simultaneously pad in a reflection fashion
+        and map to complex. 
+
+        Parameters
+        ----------
+        pad_size : list of 4 integers
+            Size of padding to apply [top, bottom, left, right].
+        input_size : list of 2 integers
+            Size of the original signal [height, width].
+        pre_pad : boolean, optional
+            If set to true, then there is no padding one simply adds the
+            imaginary part. 
+
+        """
+        self.pre_pad = pre_pad
+        self.pad_size = pad_size
+        self.input_size = input_size
+
+        self.build()
+
+    def build(self):
+        """Builds the padding module. 
+            
+            Attributes
+            ----------
+            padding_module : ReflectionPad2d
+                Pads the input tensor using the reflection of the input
+                boundary.
+
+        """
+        pad_size_tmp = list(self.pad_size)
+
+        # This handles the case where the padding is equal to the image size
+        if pad_size_tmp[0] == self.input_size[0]:
+            pad_size_tmp[0] -= 1
+            pad_size_tmp[1] -= 1
+        if pad_size_tmp[2] == self.input_size[1]:
+            pad_size_tmp[2] -= 1
+            pad_size_tmp[3] -= 1
+        if pad_size_tmp[4] == self.input_size[2]:
+            pad_size_tmp[4] -= 1
+            pad_size_tmp[5] -= 1
+
+        self.pad_size_tmp = pad_size_tmp
+
+    def __call__(self, x):
+        """Applies padding and maps to complex.
+
+            Parameters
+            ----------
+            x : tensor
+                Real tensor input to be padded and sent to complex domain.
+            Returns
+            -------
+            output : tensor
+                Complex torch tensor that has been padded.
+
+        """
+        if not self.pre_pad:
+            # Pytorch expects its padding as [left, right, top, bottom]
+            pad_size_tmp = self.pad_size_tmp
+            x = pad(x, (pad_size_tmp[2], pad_size_tmp[3], 
+                pad_size_tmp[0], pad_size_tmp[1], pad_size_tmp[4],
+                pad_size_tmp[5]), mode='constant')
+
+            # Note: PyTorch is not effective to pad signals of size N-1 with N
```

Comment:
We are padding with zeros for 3D, at least according to `mode=constant`. So this stuff should probably go along with the stuff above